python/lsst/pipe/drivers/coaddDriver.py

```diff
@@ -0,0 +1,297 @@
+        pool.storeSet(butler=butler, skymap=butler.get(self.config.coaddName + "Coadd_skyMap"))
+        tractIdList = []
+        for patchRefList in tractPatchRefList:
+            tractSet = set([patchRef.dataId["tract"] for patchRef in patchRefList])
+            assert len(tractSet) == 1
+            tractIdList.append(tractSet.pop())
+
+        selectDataList = [data for data in pool.mapNoBalance(self.readSelection, selectIdList) if
+                          data is not None]
+        nonEmptyList = pool.mapNoBalance(self.checkTract, tractIdList, selectDataList)
+        tractPatchRefList = [patchRefList for patchRefList, nonEmpty in
+                             zip(tractPatchRefList, nonEmptyList) if nonEmpty]
+        self.log.info("Non-empty tracts (%d): %s" % (len(tractPatchRefList),
+                                                     [patchRefList[0].dataId["tract"] for patchRefList in
+                                                      tractPatchRefList]))
+
+        # Install the dataRef in the selectDataList
+        for data in selectDataList:
+            data.dataRef = getDataRef(butler, data.dataId, "calexp")
+
+        # Process the non-empty tracts
+        return [self.runTract(patchRefList, butler, selectDataList) for patchRefList in tractPatchRefList]
+
+    @abortOnError
+    def runTract(self, patchRefList, butler, selectDataList=[]):
+        """Run stacking on a tract
+        This method only runs on the master node.
+        @param patchRefList: List of patch data references for tract
+        @param butler: Data butler
+        @param selectDataList: List of SelectStruct for inputs
+        """
+        pool = Pool("stacker")
+        pool.cacheClear()
+        pool.storeSet(butler=butler, warpType=self.config.coaddName + "Coadd_tempExp",
+                      coaddType=self.config.coaddName + "Coadd")
+        patchIdList = [patchRef.dataId for patchRef in patchRefList]
+
+        selectedData = pool.map(self.warp, patchIdList, selectDataList)
+        if self.config.doBackgroundReference:
+            self.backgroundReference.run(patchRefList, selectDataList)
+
+        refNamer = lambda patchRef: tuple(map(int, patchRef.dataId["patch"].split(",")))
+        lookup = dict(zip(map(refNamer, patchRefList), selectedData))
+        coaddData = [Struct(patchId=patchRef.dataId, selectDataList=lookup[refNamer(patchRef)]) for
+                     patchRef in patchRefList]
+        pool.map(self.coadd, coaddData)
+
+    def readSelection(self, cache, selectId):
+        """Read Wcs of selected inputs
+        This method only runs on slave nodes.
+        This method is similar to SelectDataIdContainer.makeDataRefList,
+        creating a Struct like a SelectStruct, except with a dataId instead
+        of a dataRef (to ease MPI).
+        @param cache: Pool cache
+        @param selectId: Data identifier for selected input
+        @return a SelectStruct with a dataId instead of dataRef
```

Comment:
As stated before, it would be nice to have parameter documentation like this on all the methods.